db/state/inverted_index.go

```diff
@@ -892,6 +893,150 @@ func (iit *InvertedIndexRoTx) prune(ctx context.Context, rwTx kv.RwTx, txFrom, t
 		}
 	}
 
+	iit.ii.logger.Info("ii pruning res", "name", iit.name, "pruned", stat.PruneCountTx)
+
+	return stat, err
+}
+
+func (iit *InvertedIndexRoTx) prune(ctx context.Context, rwTx kv.RwTx, txFrom, txTo, limit uint64, logEvery *time.Ticker, fn func(key []byte, txnum []byte) error) (stat *InvertedIndexPruneStat, err error) {
+	stat = &InvertedIndexPruneStat{MinTxNum: math.MaxUint64}
+
+	mxPruneInProgress.Inc()
+	defer mxPruneInProgress.Dec()
+	defer func(t time.Time) { mxPruneTookIndex.ObserveDuration(t) }(time.Now())
+
+	if limit == 0 { // limits amount of txn to be pruned
+		limit = math.MaxUint64
+	}
+
+	ii := iit.ii
+	ii.logger.Info("ii pruning", "name", iit.name, "txFrom", txFrom, "txTo", txTo, "limit", limit)
+
+	//defer func() {
+	//	ii.logger.Error("[snapshots] prune index",
+	//		"name", ii.filenameBase,
+	//		"forced", forced,
+	//		"pruned tx", fmt.Sprintf("%.2f-%.2f", float64(minTxnum)/float64(iit.stepSize), float64(maxTxnum)/float64(iit.stepSize)),
+	//		"pruned values", pruneCount,
+	//		"tx until limit", limit)
+	//}()
+
+	keysCursor, err := rwTx.CursorDupSort(ii.KeysTable)
+	if err != nil {
+		return stat, fmt.Errorf("create %s keys cursor: %w", ii.FilenameBase, err)
+	}
+	defer keysCursor.Close()
+	idxDelCursor, err := rwTx.RwCursorDupSort(ii.ValuesTable)
+	if err != nil {
+		return nil, err
+	}
+	defer idxDelCursor.Close()
+
+	var txKey [8]byte
+	binary.BigEndian.PutUint64(txKey[:], txFrom)
+
+	// Invariant: if some `txNum=N` pruned - it's pruned Fully
+	// Means: can use DeleteCurrentDuplicates all values of given `txNum`
+	for v, k, err := idxDelCursor.First(); k != nil; v, k, err = idxDelCursor.NextNoDup() {
```

Comment:
`v, k, err :=` let's avoid `v, k` naming - because it's not used anywhere in Erigon. Let's use `k, v` or `k, txNumBytes`